docs/source/transaction_family_specifications/track_and_trade_transaction_family.rst

```diff
@@ -0,0 +1,747 @@
+       // its associated Record. These are required to distinguish pages
+       // with colliding addresses.
+       string name = 1;
+       string record_id = 2;
+
+       // ReportedValues are sorted first by timestamp, then by reporter_index.
+       repeated ReportedValue reported_values = 4;
+   }
+
+
+Properties and PropertyPages whose addresses collide are stored in
+lists alphabetized by Property name.
+
+.. code-block:: protobuf
+
+   message PropertyContainer {
+       repeated Property entries = 1;
+   }
+
+   message PropertyPageContainer {
+       repeated PropertyPage entries = 1;
+   }
+
+.. _RecordTypes:
+
+Record Types
+------------
+
+In order to validate incoming tracking data, Records are assigned a
+RecordType at creation. A RecordType is a user-defined list of
+PropertySchemas, each of which has a name and data type.
+PropertySchemas may be designated as ``required``. A required Property
+must be initialized with a value at the time of a Record's creation.
+For example, a ``Fish`` type might list ``species`` as required, but
+not ``temperature``, since temperature wouldn't be known until
+measurements were taken. Properties not specified at Record creation
+are initialized as empty lists.
+
+.. code-block:: protobuf
+
+   message PropertySchema {
+       enum DataType {
+           BYTES = 0;
+	   STRING = 1;
+	   INT = 2;
+	   FLOAT = 3;
+	   LOCATION = 4;
+       }
+
+       // The name of the property, e.g. "temperature"
+       string name = 1;
+
+       // The Property's type (int, string, etc.)
+       DataType data_type = 2;
+
+       // A flag indicating whether initial values must be provided for the
+       // Property when a Record is created.
+       bool required = 3;
+   }
+
+
+   message RecordType {
+       // A unique human-readable designation for the RecordType
+       string name = 1;
+
+       repeated PropertySchema properties = 2;
+   }
+
+
+A Record will have all and only the Properties listed in its type, and
+new Records cannot be created without a type. As a consequence, a
+type-creation transaction must be executed before any Records can be
+created.
+
+RecordTypes whose addresses collide are stored in a list alphabetized
+by name.
+
+.. code-block:: protobuf
+
+   message RecordTypeContainer {
+       repeated RecordType entries = 1;
+   }
+
+
+Because it is expected to be used for many RecordTypes, a dedicated
+Location protobuf message is used, the values of which are latitude
+and longitude.
+
+.. code-block:: protobuf
+
+  message Location {
+        // Coordinates are expected to be in millionths of a degree
+        sint64 latitude = 1;
+        sint64 longitude = 2;
+  }
+
+
+Agents
+------
+
+Agents are entities that can send transactions affecting Records. This
+could include not only humans and companies that act as owners and
+custodians of objects being tracked, but also autonomous sensors
+sending transactions updating Records' data. All Agents must be
```

Comment:
I'd change "updating" to "that update" (or "to update") in order to avoid that second gerund.